### PR TITLE
Create and upload AMI on Express Gateway deploy — Connect LunchBadger/general#272

### PR DESCRIPTION
**Pull request by XVincentX**
This PR will add another task for our CircleCI configuration in order to create and deploy an AMI image on AWS every time we'll release a new Express Gateway version.

I've self-commented this pull request in all the parts where I'm not sure I made the best choice — please provide me some feedback on that.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ jobs:
       - run: npm install
       - run: npm test
 
-  deploy:
+  release:
```
Comment (XVincentX) [inline .circleci/config.yml]: I've renamed this to `release` because effectively we're not deploying anything.

File: .circleci/config.yml [hunk 2]
```diff
@@ -58,27 +58,27 @@ jobs:
...
-              only: /v\d+(\.\d+){2}/
-      - node-8:
-          filters:
-            tags:
-              only: /v\d+(\.\d+){2}/
-      - node-9:
-          filters:
-            tags:
-              only: /v\d+(\.\d+){2}/
-      - node-8-real-redis:
```
Comment (XVincentX) [inline .circleci/config.yml]: yaml references will dry the code a bit

File: .circleci/config.yml [hunk 3]
```diff
@@ -58,27 +58,27 @@ jobs:
...
       - run: "docker push expressgateway/express-gateway:latest"
 
+  packer:
+    docker:
+      - image: hashicorp/packer:light
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+      - run: packer build ./.circleci/packer.json
```
Comment (XVincentX) [inline .circleci/config.yml]: Do you think is a good idea to put `packer.json` in `.circleci` directory? Maybe a new directory called `release` with all Dockerfiles and `packer.json` file as well?

File: .circleci/packer.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        "owners": [
+          "137112412989",
+          "591542846629",
+          "801119661308",
+          "102837901569",
+          "013907871322",
+          "206029621532",
+          "286198878708",
+          "443319210888"
+        ],
```
Comment (XVincentX) [inline .circleci/packer.json]: The amount of base images is huge. I took the latest Amazon Linux 2 image I've been able to find, but to be honest I'm not sure whether it's the right one or not.

File: .circleci/packer.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      "ami_description": "Amazon Linux with Express Gateway"
+    }
+  ],
+  "provisioners": [
+    {
+      "type": "shell",
+      "inline": [
+        "curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.11/install.sh | bash",
+        ". ~/.nvm/nvm.sh",
+        "nvm install --lts",
```
Comment (XVincentX) [inline .circleci/packer.json]: This basically means — Node 8. I guess it's ok for us.

File: .circleci/config.yml (hunk 3 above)
Comment (kevinswiber) [reply .circleci/config.yml]: I think the `release` directory is a good idea.  Just make sure it's in `.npmignore` and `.dockerignore`.

File: .circleci/packer.json (hunk 5 above)
Comment (kevinswiber) [reply .circleci/packer.json]: This should be fine.

File: .circleci/config.yml (hunk 3 above)
Comment (XVincentX) [reply .circleci/config.yml]: Will do, thanks!

**Comment by XVincentX** [comment]
I didn't address your comments :) but thanks for merging.

**Pull request merged**